crates/oxc_language_server/src/main.rs

```diff
@@ -67,11 +67,42 @@ impl Options {
         }
     }
 
-    fn get_config_path(&self) -> Option<PathBuf> {
-        if self.config_path.is_empty() {
+    fn get_config_path(&self, root_path: &Path) -> Option<PathBuf> {
+        let Some(config_path) = &self.config_path else {
+            // no config file is provided, let search in the root_path for one
+            let search_configs = vec![
+                "oxlintrc.json",
+                "oxlint.json",
+                ".oxlintrc.json",
+                ".oxlint.json",
+                ".oxlintrc",
+                ".eslintrc",
+            ];
+
+            for search_config in search_configs {
+                let config_path = root_path.join(search_config);
+
+                if config_path.exists() {
+                    return Some(config_path);
+                }
+            }
+
+            return None;
+        };
+
+        // config path is provided and is empty, e.g "oxc.configPath": ""
+        if config_path.is_empty() {
             None
         } else {
-            Some(PathBuf::from(&self.config_path))
+            // provided config path is not empty
+            let config_path = root_path.join(config_path);
+
+            if config_path.exists() {
+                return Some(config_path);
+            }
+
+            warn!("could not found oxlint config in {:?}", config_path);
+            None
```

Comment:
The CLI defaults to `oxlintrc.json`, should that be done here as well instead of nothing if a config isn't found?

Might also be worthwhile to enhance the CLI with the autodetection ability.

Reply:
In this branch the user defines manually a config path, I would not try to find a config

Reply:
I think I left this at the wrong spot, I meant for it to be with the searching for different config files.

It wouldn’t make sense there either though because you would have already looked for this file and failed before returning none.

Reply:
I don’t think I have access to resolve this comment, but I’m good here.

Reply:
`oxlintrc.json` is the first file to be searched.   

I do not know if I want report an not found config (in search branch),  
because some projects have no config and using CLI only.

Please explain more what you want to see changed :)